components/dialog/test/dialog-fullscreen.vdiff.js

```diff
@@ -0,0 +1,79 @@
+import '../dialog-fullscreen.js';
+import { expect, fixture, html, nextFrame } from '@brightspace-ui/testing';
+import { footer, general, long, tabs } from './dialog-shared-contents.js';
+import { ifDefined } from 'lit/directives/if-defined.js';
+
+function dialog(opts) {
+	const defaults = { content: html`${general}${footer}`, noPadding: false };
+	const { content, noPadding, width } =  { ...defaults, ...opts };
+	return html`
+		<d2l-dialog-fullscreen title-text="Dialog Title" ?no-padding="${noPadding}" width="${ifDefined(width)}" opened>
+			${content}
+		</d2l-dialog-fullscreen>
+	`;
+}
+
+function dispatchFullscreenWithinEvent(elem, state) {
+	elem.dispatchEvent(new CustomEvent(
+		'd2l-fullscreen-within', { bubbles: true, composed: true, detail: { state: state } }
+	));
+}
+
+describe('dialog-fullscreen', () => {
+
+	['native', 'custom'].forEach((type) => {
+
+		describe(type, () => {
+			before(async() => {
+				window.D2L.DialogMixin.preferNative = type === 'native';
+			});
+
+			[
+				{ screen: 'wider', viewport: { width: 1400, height: 700 } },
+				{ screen: 'wide', viewport: { width: 800, height: 500 } },
+				{ screen: 'narrow', viewport: { width: 600, height: 500 } },
+				{ screen: 'landscape', viewport: { width: 600, height: 320 } }
+			].forEach(({ screen, viewport }) => {
+				describe(screen, () => {
+					[
+						{ name: 'opened', f: dialog() },
```

Comment:
Nit: The single letter property gives me vibes from the olden days when I was doing Fortran limited to 640K. Lol. Unless it's ubiquitous, I think I would always prefer writing it out for ease of readability. Not a big deal though.